File: test/js/natchMatcherTest.js
import NatchMatcher from '../../src/js/natchMatcher';

const assert = require('assert');
const fs = require('fs');
const path = require('path');

/* eslint-env mocha */
describe('Natural Language Typeahead Matcher', () => {
  const fixturesDir = path.join(__dirname, '../fixtures');
  const testData = JSON.parse(fs.readFileSync(path.join(fixturesDir, 'tests.json')).toString());
  const subjects = testData.subjects;
  const tests = testData.tests;

  for (let test of tests) {
    it(test.message, () => {
      const matcher = new NatchMatcher(test.query);

      assert.deepEqual(test.expectedResult, matcher.filter(subjects));
    });
  }
});

File: test/python/natch_matcher_test.py
# -*- coding: utf-8 -*-
"""
:copyright: (c) 2016 by Zachary Ozer
"""
import json
import os
import unittest

import natch_matcher


class TestNatchMatcher(unittest.TestCase):

  def test(self):
    f = open(os.path.join(os.path.dirname(__file__), '../fixtures/tests.json'))
    json_data = json.loads(f.read())
    subjects = json_data.get('subjects')
    tests = json_data.get('tests')
    for test in tests:
      message = test['message']
      query = test['query']
      expected_result = test['expectedResult']
      matcher = natch_matcher.NatchMatcher(query)
      self.assertEquals(expected_result, matcher.filter(subjects), message)

if __name__ == '__main__':
  unittest.main()
